ansible/roles/operator-pipeline/templates/openshift/tasks/acquire-lock.yaml

```diff
@@ -31,24 +27,44 @@ spec:
             else
                 TOTAL_DURATION_IN_SECONDS=${1}
             fi
+            export TOTAL_DURATION_IN_SECONDS
         }
         acquire_lease() {
             echo "Attempting to create lease $(params.lease-name)"
-            until kubectl create -f e2e-lease.yaml
+            while true
             do
-              echo "Waiting for lease..."
-              acquiredTime=$(oc get lease $(params.lease-name) -o=jsonpath='{.spec.acquireTime}')
-              acquiredTimeInSeconds=$(date +%s --date="${acquiredTime}")
-              currentTimeinSeconds=$(date +%s)
-              elapsedSeconds=$(($currentTimeinSeconds-$acquiredTimeInSeconds))
-              if (( elapsedSeconds > TOTAL_DURATION_IN_SECONDS)); then
-                echo "Lease abandoned, deleting"
-                kubectl delete lease $(params.lease-name)
+              if kubectl create -f e2e-lease.yaml; then
+                leaseOwner=$(kubectl get lease $(params.lease-name) -o=jsonpath='{.spec.holderIdentity}')
+                echo "The owner of the lease is ${leaseOwner}"
+                echo "Acquired lease $(params.lease-name)"
+                break
+              else
+                echo "Waiting for lease..."
+                acquiredTime=$(oc get lease $(params.lease-name) -o=jsonpath='{.spec.acquireTime}')
+                echo "acquiredTime: ${acquiredTime}"
+                acquiredTimeInSeconds=$(date +%s --date="${acquiredTime}")
+                echo "acquiredTimeInSeconds: ${acquiredTimeInSeconds}"
+                c=$(date +"%D %T")
+                echo "currenttime: ${c}"
+                currentTimeinSeconds=$(date +%s)
+                echo "currentTimeinSeconds: ${currentTimeinSeconds}"
+                elapsedSeconds=$(($currentTimeinSeconds-$acquiredTimeInSeconds))
+                echo "TOTAL_DURATION_IN_SECONDS: ${TOTAL_DURATION_IN_SECONDS}"
+                echo "elapsedSeconds: ${elapsedSeconds}"
+                if (( elapsedSeconds > TOTAL_DURATION_IN_SECONDS)); then
+                  echo "Lease abandoned, deleting"
+                  kubectl delete lease $(params.lease-name)
+                  kubectl create -f e2e-lease.yaml
+                  leaseOwner=$(kubectl get lease $(params.lease-name) -o=jsonpath='{.spec.holderIdentity}')
+                  echo "The owner of the lease is ${leaseOwner}"
+                  echo "Acquired lease $(params.lease-name)"
+                  break
```

Comment:
This can be dropped as the same action will be executed at the start of the next loop.